src/Common/src/System/Net/InteropIPAddressExtensions.Unix.cs

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace System.Net
+{
+    internal static class InteropIPAddressExtensions
+    {
+        public static unsafe IPAddress GetIPAddress(this Interop.Sys.IPAddress nativeIPAddress)
+        {
+            if (!nativeIPAddress.IsIPv6)
```

Comment:
I originally had this as `if (nativeIPAddress.IsIPv6 == 0)` when the relevant field was integral-typed.